devito/ir/support/basic.py

```diff
@@ -457,6 +468,9 @@ def __eq__(self, other):
             isinstance(other, TimedAccess) and\
             self.directions == other.directions
 
+    def __hash__(self):
```

Comment:
@tjb900 do we actually need to override __hash__ 4 times? is this due to multiple inheritance?

Reply:
discussed offline, all good